src/plugins/util/fileUploader.js

```diff
@@ -173,38 +177,97 @@ const fileUploader = (() => {
         uploadErrorSection.appendChild(uploadErrorTitle);
         uploadErrorSection.appendChild(uploadErrorInfoDiv);
 
+        // Create success section (initially hidden)
+        const uploadSuccessSection = document.createElement('div');
+        uploadSuccessSection.className = 'gm-section gm-upload-success hidden';
+
+        const uploadSuccessInfoDiv = document.createElement('div');
+        uploadSuccessInfoDiv.className = 'gm-info-flex';
+
+        const uploadSuccessTitle = document.createElement('div');
+        uploadSuccessTitle.className = 'gm-uploading-file-title';
+        uploadSuccessTitle.innerHTML = 'Uploading files';
+
+        const uploadSuccessIcon = document.createElement('i');
+        uploadSuccessIcon.className = 'gm-downloadFile-icon';
+
+        const uploadSuccessText = document.createElement('div');
+        uploadSuccessText.className = 'gm-success-text';
+
+        const completedInfoDiv = document.createElement('div');
+        completedInfoDiv.className = 'gm-info-flex gm-completed-text';
+
+        const completedIcon = document.createElement('i');
+        completedIcon.className = 'gm-check-icon';
+
+        const completedText = document.createElement('div');
+        completedText.innerHTML = i18n.COMPLETED || 'Completed';
+
+        completedInfoDiv.appendChild(completedText);
+        completedInfoDiv.appendChild(completedIcon);
+
+        uploadSuccessInfoDiv.appendChild(uploadSuccessIcon);
+        uploadSuccessInfoDiv.appendChild(uploadSuccessText);
+        uploadSuccessInfoDiv.appendChild(completedInfoDiv);
+
+        uploadSuccessSection.appendChild(uploadSuccessTitle);
+        uploadSuccessSection.appendChild(uploadSuccessInfoDiv);
+
         // Add all sections to container
         container.appendChild(dragDropArea);
         container.appendChild(uploadProgressSection);
         container.appendChild(uploadErrorSection);
+        container.appendChild(uploadSuccessSection);
+
+        const getFileNameToDisplay = (name) => {
+            const truncateThreshold = 30;
+
+            let fileName = name;
+            if (fileName.length > truncateThreshold) {
+                const [nameWithoutExt, extension = ''] = fileName.split(/(?=\.[^.]+$)/);
+                fileName =
+                    (nameWithoutExt.length > 15 ? `${nameWithoutExt.slice(0, 30)}...` : nameWithoutExt) + extension;
```

Comment:
Two questions here:

1. Is this `15` correct? Why not `30`?

2. Why not use our constant here in the slice: `slice(0, truncateThreshold)`?